docs/how-to-run-bench-gimlet.adoc

```diff
@@ -0,0 +1,239 @@
+data_link = "rge1"
+----
+
+Next, setup your omicron package target.  I'm using the name "sn21" for my target:
+[source,text]
+----
+./target/release/omicron-package -t sn21 target create -i standard -m non-gimlet -s softnpu
+----
+
+Then make the packages:
+[source,text]
+----
+./target/release/omicron-package --target sn21 package
+----
+
+Once completed, copy these files over to the gimlet:
+[source,text]
+----
+rsync --delete -Paz tools out \
+  ./target/release/omicron-package package-manifest.toml \
+  --exclude /out/downloads/netstack.* \
+  --exclude *.p5p \
+  root@gimlet-sn21:/tmp/omicron
+----
+
+This includes the `tools` directory as we make use of a few scripts.
+
+== On the gimlet.
+
+After copying your Omicron packages over, setup the system with "virtual hardware".
+In my example, the `rge1` is the network interface the gimlet will be using.
+All commands on the gimlet are run as root.
+
+[source,text]
+----
+cd /tmp
+$ PHYSICAL_LINK=rge1 ./tools/create_virtual_hardware.sh
+----
+
+== Install Omicron
+Next, we install Omicron
+
+[source,console]
+----
+cd /tmp
+./omicron-package -t sn21 install
+----
+
+You should start seeing zones appear, and you can monitor the sled-agent log
+for progress with `tail -f $(svcs -L sled-agent)`.  It will take a few seconds
+for the sled-agent service to start.
+
+Watch for the oxz_switch zone to appear, and give it an additional 20/30 seconds
+to come online.  Then, run this command to setup softnpu.
+The gateway IP is the I.P. address of the gateway for the omicron network.
+The gateway MAC is the mac address of the gateway.
+
+[source,console]
+----
+export GATEWAY_IP=172.20.24.1
+export GATEWAY_MAC=aa:0:4:0:ca:fe
+tools/scrimlet/softnpu-init.sh
+----
+
+=== Additional configuration after starting Omicron
+
+Internal services that require external connectivity (e.g. Nexus, Boundary NTP,
+External DNS) do so via OPTE. When using SoftNPU we'll need to configure Proxy ARP for
+the services IP Pool.  Use the values from your [rack_network_config] section of
+the config-rss.toml you setup above.
+
+[source,console]
+----
+# dladm won't return leading zeroes but `scadm` expects them
+# Look at your `dladm show-vnic sc0_1 -p -o macaddress` output, get the mac address
+# and add any leading zeros to octets that require it.
+
+export SOFTNPU_MAC=a8:e1:de:01:70:1d
+
+/opt/oxide/softnpu/stuff/scadm \
+  --server /opt/oxide/softnpu/stuff/server \
+  --client /opt/oxide/softnpu/stuff/client \
+  standalone \
+  add-proxy-arp \
+  $SERVICE_IP_POOL_START \
+  $SERVICE_IP_POOL_END \
+  $SOFTNPU_MAC
+----
+
+You'll also need to configure Proxy ARP.  The ip pool start and ip pool end also
+come from your config-rss.toml
+[source,console]
+----
+export SOFTNPU_MAC=a8:e1:de:01:70:1d
+
+opt/oxide/softnpu/stuff/scadm \
+  --server /opt/oxide/softnpu/stuff/server \
+  --client /opt/oxide/softnpu/stuff/client \
+  standalone \
+  add-proxy-arp \
+  $IP_POOL_START \
+  $IP_POOL_END \
+  $SOFTNPU_MAC
```

Comment:
So in the case where an entire subnet is allocated to us and the upstream router is configured to route said subnet to us, we don't actually need to configure proxy arp (*) for the ip pools.

(*) that is except for `uplink_ip` itself (on a real sidecar we add the address on the qsfp0 link in the switch zone and so ARPs for it get handled normally).